### DISCUSSION TITLE
B. Bad Boy

### DISCUSSION THREAD

**Discussion by MnG-Official** [sonsuzus/sonsuzus#525] [Zeka Sorusu] [answered]
Riley çok kötü bir çocuktur, ama aynı zamanda bir yoyo ustasıdır. Bu yüzden yoyo becerilerini arkadaşı Anton'ı sinirlendirmek için kullanmaya karar verir.

Anton'un odası _n_ satır ve _m_ sütundan oluşan bir ızgara olarak gösterilebilir. _(i,j)_ satır _i_ ve sütun _j_'deki hücreyi göstersin. Anton şu anda odasında _(i,j)_ konumunda durmaktadır. Anton'u sinirlendirmek için Riley odadaki hücrelere tam olarak iki yoyo atmaya karar verir (aynı hücrede olabilirler).

Anton yere atılan yoyolardan hoşlanmadığı için ikisini de alıp başlangıç ​​konumuna geri dönmek zorundadır. Anton'un kat ettiği mesafe, her iki yoyo'nun da konumlarından geçen ve yalnızca yan yana bitişik hücrelere giderek _(i,j)_ konumuna geri dönen en kısa yoldur. Yani, _(x,y)_ hücresindeyse, _(x+1,y),_ _(x−1,y),_ _(x,y+1)_ ve _(x,y−1)_ hücrelerine tek adımda gidebilir (eğer bu koordinatlara sahip bir hücre varsa).

Riley, Anton'ın kat ettiği mesafeyi en üst düzeye çıkarmak için bu iki yoyoyu nereye atması gerektiğini merak ediyor. Fakat çok meşgul olduğu için, bunu ona söylemenizi istedi.

### **INPUT**
İlk satır, tek bir tam sayı _t_ _(1≤t≤104)_ içerir — test vakalarının sayısı. Ardından t test vakası gelir.

Her test vakasının tek satırı, odanın boyutları ve Anton'ın şu anda durduğu hücre olan dört tam sayı _n,_ _m,_ _i,_ _j_ _(1≤n,m≤109,_ _1≤i≤n,_ _1≤j≤)_ içerir.

### **OUTPUT**
Her test durumu için, dört tam sayı _x1,_ _y1,_ _x2,_ _y2_ _(1≤x1,x2≤n,_ _1≤y1,y2≤m)_ yazdırın — iki yoyo'nun atılması gereken koordinatlar. _(x1,y1)_ ve _(x2,y2)_ koordinatlarına atılacaklar.

Birden fazla cevap varsa, herhangi birini yazdırabilirsiniz.

### **Örnek**
**Giriş**
7
2 3 1 1
4 4 1 2
3 5 2 2
5 1 2 1
3 1 3 1
1 1 1 1
1000000000 1000000000 1000000000 50
**Çıktı**
1 2 2 3
4 1 4 4
3 1 1 5
5 1 1 1
1 1 2 1
1 1 1 1
50 1 1 1000000000

**Not**
İşte ilk test vakasının görselleştirilmesi.
![image](https://github.com/user-attachments/assets/2ac0c179-dcb7-477d-87d6-f1c0ffb90b01)

**Comment by MnG-Official** [comment, marked as answer]
```C
#include <stdio.h>

void solve(long long n_uzunluk, long long m_uzunluk, long long i_uzunluk, long long j_uzunluk) {
    printf("1 1 %lld %lld\n", n_uzunluk, m_uzunluk);
}

int main() {
    int t;
    printf("Test vakasının sayısı: ");
    scanf("%d", &t);

    for (int k = 0; k < t; k++) {
        long long n_uzunluk, m_uzunluk, i_uzunluk, j_uzunluk;
        scanf("%lld %lld %lld %lld", &n_uzunluk, &m_uzunluk, &i_uzunluk, &j_uzunluk);
        solve(n_uzunluk, m_uzunluk, i_uzunluk, j_uzunluk);
  }

return 0;
}
```

**Reply by sonsuzus** [reply]
Daha kısa çözülebilirdi. :) Tebrikler.

**Reply by sonsuzus** [reply]
Ayrıca cumartesi günü son yarım saati unutmuş değilim. 🥲

**Reply by MnG-Official** [reply]
> Ayrıca cumartesi günü son yarım saati unutmuş değilim. 🥲

Söz bu hafta daha iyileriyle geleceğim. Size daha yaraşır birtakım şeylerle...
(O hafta çok araştıracak vaktim yoktu o yüzden iki hafta önceninkini göstermiştim size. Normalde benim bir çizelgem var hangi hafta ne olabilir diye.)
